components/table/table-col-sort-button.js

```diff
@@ -108,20 +119,57 @@ export class TableColSortButton extends LocalizeCoreElement(FocusMixin(LitElemen
 		const buttonTitle = this.nosort
 			? undefined
 			: this.localize('components.table-col-sort-button.title', {
-				sourceType: this.sourceType,
-				direction: this.desc ? 'desc' : undefined
+				sourceType: this._hasDropdownItems && this._selectedMenuItemText ? 'value' : this.sourceType,
+				direction: this.desc ? 'desc' : undefined,
+				selectedMenuItemText: this._selectedMenuItemText
 			});
 
 		const iconView = !this.nosort ?
 			html`<d2l-icon icon="${this.desc ? 'tier1:arrow-toggle-down' : 'tier1:arrow-toggle-up'}"></d2l-icon>` :
 			null;
 
-		return html`<button
+		const button = html`<button
 				aria-describedby="${this._describedById}"
+				class="${classMap({ 'd2l-dropdown-opener': this._hasDropdownItems })}"
 				title="${ifDefined(buttonTitle)}"
 				type="button">
 				<slot></slot>${iconView}
 			</button><span id="${this._describedById}" hidden>${buttonDescription}</span>`;
+		if (this._hasDropdownItems) {
+			return html`<d2l-dropdown>
+					${button}
+					<d2l-dropdown-menu no-pointer align="start" vertical-offset="0">
+						<d2l-menu @d2l-menu-item-change="${this._handleMenuItemChange}">
+							<slot name="items" @slotchange="${this._handleSlotChange}"></slot>
+						</d2l-menu>
+					</d2l-dropdown-menu>
+				</d2l-dropdown>`;
+		} else {
+			return html`${button}<slot name="items" @slotchange="${this._handleSlotChange}"></slot>`;
+		}
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		// de-select any selected dropdown menu item
+		if (changedProperties.has('nosort') && this.nosort && this._hasDropdownItems) {
+			const selectedItem = this.querySelector('[selected]');
+			if (selectedItem) selectedItem.selected = false;
+		}
```

Comment:
I'm torn on this bit as we've generally been having the consumer do everything with this (e.g., setting `nosort` and `desc`) so I could see arguing that the consumer should handle this as well.